lib/resolve.js

```diff
@@ -29,6 +29,17 @@ module.exports = unyield(function* resolve(slug, options) {
   if (!parsed.ref) parsed.ref = '*';
   var ref = parsed.ref;
 
+  if (isSHA(ref)) {
+    debug('%s: resolving via commit SHA', slug, ref);
+    try {
```

Comment:
because `get()` throws if it can't resolve something.  if we just threw that error and exited, we'd introduce another regression: `require('foo/bar@branch-that-looks-like-an-sha')`